noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/private_call_data_validator.nr

```diff
@@ -144,37 +148,27 @@ impl PrivateCallDataValidator {
         );
     }
 
-    pub fn validate_against_call_request(self, scoped_call_request: ScopedPrivateCallRequest) {
+    pub fn validate_against_call_request(self, request: PrivateCallRequest) {
         let call_stack_item = self.data.call_stack_item;
-        let caller_contract_address = scoped_call_request.contract_address;
-        let request = scoped_call_request.call_request;
 
-        assert(
-            request.matches_stack_item(call_stack_item), "calculated private_call_hash does not match provided private_call_hash at the top of the call stack"
+        assert_eq(
+            request.contract_address, call_stack_item.contract_address, "contract_address does not match call request"
+        );
+        assert_eq(
+            request.call_context, call_stack_item.public_inputs.call_context, "call_context does not match call request"
+        );
+        assert_eq(
+            request.args_hash, call_stack_item.public_inputs.args_hash, "args_hash does not match call request"
+        );
+        assert_eq(
+            request.returns_hash, call_stack_item.public_inputs.returns_hash, "returns_hash does not match call request"
+        );
+        assert_eq(
+            request.start_side_effect_counter, call_stack_item.public_inputs.start_side_effect_counter, "start_side_effect_counter does not match call request"
+        );
+        assert_eq(
+            request.end_side_effect_counter, call_stack_item.public_inputs.end_side_effect_counter, "end_side_effect_counter does not match call request"
         );
-
-        let call_context = call_stack_item.public_inputs.call_context;
-        let caller_context = request.caller_context;
-        let is_caller_hidden = caller_context.is_hidden();
-
-        if call_context.is_delegate_call {
-            assert(!is_caller_hidden, "caller context cannot be hidden for delegate calls");
```

Comment:
I don't see the `is_hidden` check replicated in the new checks - is it covered by `target_context.msg_sender == this_context.msg_sender` or do we still need an is zero check?

Reply:
`is_hidden` is for checking the caller_context (not to be confused with call context). And caller_context has been removed.

### Before

[In private function]
A private function sets its own context as the caller_context in each call request it's making. If it's making a delegate call, it must provide its msg_sender and storage_contract_address to that call request. If not, then the values can be empty.

[Kernel <> call requests]
The kernel will verify that the values in the caller_context in each call request are empty or match the caller.

[Kernel <> popped call request]
And when verifying a call, the kernel will compare the caller_context in the popped call request, with the information of the current call:
- If it's a delegate call, the current call's msg_sender and storage_contract_address must match those in the caller context in the call request. 
- If it's a normal call, then msg_sender is the (scoped) contract address of the call request, which is a value inserted by the kernel. 
We check that the caller_context is empty for normal call because we don't want to reveal the caller's msg_sender and storage_contract_address to public. (Not a problem for private call but we checked it just for consistency.)

### Now

[In private function]
A private function defines the call context in each call request directly.

[Kernel <> call requests]
The check that was done for the current call against the popped call request is now being checked for all the call requests emitted from a call. A caller doesn't have to provide it's context anymore because the values are correctly set for each call request.

[Kernel <> popped call request]
When verifying a call against the popped call request, the kernel only needs to check that the call context in the call request matches the current call's call context.


Hope this makes sense!

Reply:
Understood, thanks! I see the zero case and other cases are now separate entirely